internal/topic/core_test.go

```diff
@@ -547,3 +550,178 @@ func TestCore_Get(t *testing.T) {
 		})
 	}
 }
+
+func getMockAdmin(ctrl *gomock.Controller) *messagebrokermock.MockAdmin {
+	mockAdmin := messagebrokermock.NewMockAdmin(ctrl)
+	mockAdmin.EXPECT().DeleteTopic(gomock.Any(), gomock.Any()).
+		Return(messagebroker.DeleteTopicResponse{}, nil).AnyTimes()
+	return mockAdmin
+}
+
+func TestCore_DeleteSubscriptionTopic(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	repo := topicrepomock.NewMockIRepo(ctrl)
+	projectCore := projectcoremock.NewMockICore(ctrl)
+	brokerStore := brokerstoremock.NewMockIBrokerStore(ctrl)
+	ctx := context.Background()
+	extracteProjectID := "test-project"
+	topic := "projects/test-project/topics/test-topic"
+	extractedTopic := "test-topic"
+	type fields struct {
+		repo        IRepo
+		projectCore project.ICore
+		brokerStore brokerstore.IBrokerStore
+	}
+	type args struct {
+		ctx context.Context
+		m   *Model
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Delete Borker Topics success",
+			fields: fields{
+				repo:        repo,
+				projectCore: projectCore,
+				brokerStore: brokerStore,
+			},
+			args: args{
+				ctx: ctx,
+				m:   getTopicModelWithInput(topic, extractedTopic, extracteProjectID),
+			},
+			wantErr: false,
+		},
+		{
+			name: "Delete Borker Retry Topics ",
+			fields: fields{
+				repo:        repo,
+				projectCore: projectCore,
+				brokerStore: brokerStore,
+			},
+			args: args{
+				ctx: ctx,
+				m:   getTopicModelWithInput(topic+"-retry", extractedTopic+"-retry", extracteProjectID),
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Core{
+				repo:        tt.fields.repo,
+				projectCore: tt.fields.projectCore,
+				brokerStore: tt.fields.brokerStore,
+			}
+			projectCore.EXPECT().ExistsWithID(gomock.Any(), tt.args.m.ExtractedProjectID).DoAndReturn(func(arg0 context.Context, arg1 string) (bool, error) {
+				if arg1 == extracteProjectID {
+					return true, nil
+				} else {
+					return false, errors.New("ProjectID not found")
+				}
+			}).AnyTimes()
+			projectCore.EXPECT().Exists(tt.args.ctx, gomock.Any()).Return(true, nil).AnyTimes()
+			brokerStore.EXPECT().IsTopicCleanUpEnabled(gomock.Any()).Return(true)
+			repo.EXPECT().Delete(gomock.Any(), tt.args.m).Return(nil).AnyTimes()
```

Comment:
Assert DeleteTopic with the right topic name in the tests.

Reply:
@subin already written test cases for the right topic name please check line no. 586